aws/cloudtrail/test_cloudtrail_enabled_in_all_regions.py

```diff
@@ -0,0 +1,20 @@
+import pytest
+
+from conftest import botocore_client
+
+from aws.cloudtrail.resources import cloudtrails
+
+
+@pytest.mark.cloudtrail
+@pytest.mark.parametrize('aws_region',
+                         botocore_client.regions,
+                         ids=lambda region: region)
+def test_cloudtrail_enabled_in_all_regions(aws_region):
+    """
+    Tests that all regions have an associated cloudtrail or that there is a cloudtrail for all regions.
+    """
+    trails = cloudtrails()
```

Comment:
Yeah, as you suspected this and the EC2 test are kind of weird to me.

I guess an unwritten goal is to avoid IO in the tests, so they can be pure functions on data. Though this isn't set in stone.

Is the fetch so we never use cached data for these endpoints? If so, I think there's still a use_cache kwd arg in to the client we could try or we could build a top level non-caching client to call instead.

Reply:
I took another look at this today and realized I was just being dumb. I wanted to make sure that these tests always ran, even when there were 0 cloudtrails or 0 VPC flow logs, but in testing them along side regions / vpcs this issue is resolved. Just pushed up a fix.